ui/lib/utils/sentryHelpers.ts

```diff
@@ -0,0 +1,101 @@
+import * as url from 'url'
+import { AxiosInstance } from 'axios'
+import * as Sentry from '@sentry/react'
+
+const firstMountTransaction = Sentry.startTransaction({ name: 'first mount' })
+const transactionHub = new Map<string, typeof firstMountTransaction>()
+
+export const sentryEnabled = process.env.REACT_APP_SENTRY_ENABLED === 'true'
+
+export function markStart(name: string, op?: string) {
+  const transaction = Sentry.startTransaction({ name, op })
+  transactionHub.set(transaction.name, transaction)
+  transactionHub.set(transaction.traceId, transaction)
+  return transaction
+}
+
+export function markEnd(name: string, traceId?: string) {
+  const transaction = traceId
+    ? transactionHub.get(traceId)
+    : transactionHub.get(name)
+  if (transaction) {
+    transaction.finish()
+    transactionHub.delete(name)
+    if (traceId) {
+      transactionHub.delete(traceId)
+    }
+  }
+}
+
+export function markTag(key: string, value: string | number, traceId: string) {
+  const transaction = transactionHub.get(traceId)
+  if (transaction) {
+    transaction.setTag(key, value)
+  }
+}
+
+export const reportError: typeof Sentry.captureException = (...args) => {
+  if (sentryEnabled) {
+    return Sentry.captureException(...args)
+  }
+  return ''
+}
+
+export function initSentryRoutingInstrument() {
+  window.addEventListener('single-spa:first-mount', () => {
+    firstMountTransaction.finish()
+  })
+
+  window.addEventListener('single-spa:before-routing-event', (e: any) => {
+    const { pathname: newUrlPath } = url.parse(e.detail.newUrl)
+    const { pathname: oldUrlPath } = url.parse(e.detail.oldUrl)
```

Comment:
Because we use the hash route, not sure whether this can parse correctly.

Reply:
fixed